tests/sdk.test.ts

```diff
@@ -0,0 +1,286 @@
+            const entityDetails = await client.entity.getEntity(firstEntity.id);
+            expect(entityDetails.label).toEqual(firstEntity.label)
+            expect(entityDetails.degree).toEqual(firstEntity.degree)
+            expect(entityDetails.pep).toEqual(firstEntity.pep)
+            expect(entityDetails.psaCount).toEqual(firstEntity.psaCount)
+            expect(entityDetails.type).toEqual(firstEntity.type)
+
+            done = true
+        }
+
+    }, longTimeout);
+
+    test("resolution", async () => {
+        let done = false;
+        while (!done) {
+            const randString = generateRandomString(3)
+
+            const resolution = await client.resolution.resolution({name: randString});
+            if (resolution.data.length == 0) {
+                continue
+            }
+            expect(resolution.data.length).toBeGreaterThan(0)
+            expect(resolution.fields.name).toEqual([randString])
+
+            done = true
+        }
+    }, longTimeout);
+
+    test("records", async () => {
+        let done = false;
+        while (!done) {
+            const randString = generateRandomString(3)
+
+            // Test POST
+            const records = await client.search.searchRecord({q: randString});
+            if (records.data.length == 0) {
+                continue
+            }
+            expect(records.data.length).toBeGreaterThan(0)
+
+            // Test GET
+            const recordsGet = await client.search.searchRecordGet({q: randString});
+            expect(recordsGet.data.length).toBeGreaterThan(0)
+
+            // results should match
+            expect(records.data.length).toEqual(recordsGet.data.length)
+
+            // Use first record for testing
+            const firstRecord = records.data[0]
+            console.log(firstRecord.id)
+            console.log(firstRecord.label)
+
+            // Get record details
+            const recordDetails = await client.record.getRecord(firstRecord.id);
+            expect(recordDetails.label).toEqual(firstRecord.label)
+            expect(recordDetails.source).toEqual(firstRecord.source)
+            expect(recordDetails.publicationDate).toEqual(firstRecord.publicationDate)
+            expect(recordDetails.recordUrl).toEqual(firstRecord.recordUrl)
+            expect(recordDetails.referencesCount).toEqual(firstRecord.referencesCount)
+            expect(recordDetails.sourceUrl).toEqual(firstRecord.sourceUrl)
+
+            done = true
+        }
+    }, longTimeout);
+
+    // Test traversals
+    test("traversal", async () => {
+        let done = false;
+        while (!done) {
+            const randString = generateRandomString(3)
+
+            // get an entity
+            const entitySearchResults = await client.search.searchEntity({q: randString});
+            if (entitySearchResults.data.length == 0) {
+                console.log("no result, will retry with a different search term")
+                continue
+            }
```

Comment:
This section seems to be repeated, so could move to a function like 
```
async function runAndRetry(func) {
  while(true) {
    const randString = generateRandomString(3)
    result = await func({q: randString})
    if (result.data.length === 0) {
      console.log("no result, will retry with a different search term")
      continue
    }
    break
  }
  return result
}
```
and call 
```
const entitySearchResults = await runAndRetry(client.search.searchEntity)
```

Reply:
I have tried a bunch of approaches to this without any luck. The closest I can was this, but when I called it I got complaints about properties on the client being undefined....